### ISSUE TITLE
Broken install with pypy3 / 3.5 and python-snappy 0.5.2

### ISSUE THREAD

**Issue by behackett** [andrix/python-snappy#69]
Version 0.5.1 can be installed for pypy3 or pypy3.5, but 0.5.2 installs in a broken state.

```
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pip install python-snappy==0.5.1
Collecting python-snappy==0.5.1
  Downloading https://files.pythonhosted.org/packages/4c/8f/09f0a11dbcaedf9b0758a37ab2bb77d6c34b9c29afc4d0440019c152d2af/python-snappy-0.5.1.tar.gz
Requirement already satisfied: cffi in /opt/python/pypy3.5/lib_pypy (from python-snappy==0.5.1) (1.11.5)
Building wheels for collected packages: python-snappy
  Running setup.py bdist_wheel for python-snappy ... done
  Stored in directory: /home/ubuntu/.cache/pip/wheels/b8/2a/3d/528db6296278d2660142b513e2923177e6b1ac19a71021d1c3
Successfully built python-snappy
Installing collected packages: python-snappy
Successfully installed python-snappy-0.5.1
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pypy3
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Apr 27 2018, 15:39:57)
[PyPy 6.0.0 with GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>> import snappy
>>>> 
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pip uninstall python-snappy
Uninstalling python-snappy-0.5.1:
  Would remove:
    /home/ubuntu/snappypypy35/site-packages/python_snappy-0.5.1.dist-info/*
    /home/ubuntu/snappypypy35/site-packages/setup.py
    /home/ubuntu/snappypypy35/site-packages/snappy.py
    /home/ubuntu/snappypypy35/site-packages/snappy_cffi.py
    /home/ubuntu/snappypypy35/site-packages/test_snappy.py
    /home/ubuntu/snappypypy35/site-packages/test_snappy_cffi.py
Proceed (y/n)? y
  Successfully uninstalled python-snappy-0.5.1
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pip install python-snappy
Collecting python-snappy
Requirement already satisfied: cffi>=1.0.0 in /opt/python/pypy3.5/lib_pypy (from python-snappy) (1.11.5)
Installing collected packages: python-snappy
Successfully installed python-snappy-0.5.2
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pypy3
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Apr 27 2018, 15:39:57)
[PyPy 6.0.0 with GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>> import snappy
Traceback (most recent call last):
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy.py", line 47, in <module>
    from ._snappy import UncompressError, compress, decompress, \
ImportError: No module named 'snappy._snappy'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/snappypypy35/site-packages/snappy/__init__.py", line 1, in <module>
    from .snappy import (
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy.py", line 50, in <module>
    from .snappy_cffi import UncompressError, compress, decompress, \
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy_cffi.py", line 1, in <module>
    from _snappy_cffi import ffi, lib
ImportError: No module named '_snappy_cffi'
```

pypy (based on Python 2.7) seems to work fine.

**Comment by martindurant** [comment]
My reading of the above, is that pip did not attempt to build the package in the latter installation. Perhaps setup.py is incorrect, or there is leftover state from the previous install?

**Comment by behackett** [comment]
Installing in a fresh virtualenv doesn't help:
```
ubuntu@ip-10-122-30-65:~$ rm -rf snappypypy35
ubuntu@ip-10-122-30-65:~$ /opt/python/pypy3.5/bin/virtualenv snappypypy35
Using base prefix '/opt/python/pypy3.5'
New pypy executable in /home/ubuntu/snappypypy35/bin/pypy3
Also creating executable in /home/ubuntu/snappypypy35/bin/pypy
Installing setuptools, pip, wheel...done.
ubuntu@ip-10-122-30-65:~$ . snappypypy35/bin/activate
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pip install python-snappy
Collecting python-snappy
Requirement already satisfied: cffi>=1.0.0 in /opt/python/pypy3.5/lib_pypy (from python-snappy) (1.11.5)
Installing collected packages: python-snappy
Successfully installed python-snappy-0.5.2
(snappypypy35) ubuntu@ip-10-122-30-65:~$ pypy3
Python 3.5.3 (fdd60ed87e941677e8ea11acf9f1819466521bf2, Apr 27 2018, 15:39:57)
[PyPy 6.0.0 with GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>> import snappy
Traceback (most recent call last):
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy.py", line 47, in <module>
    from ._snappy import UncompressError, compress, decompress, \
ImportError: No module named 'snappy._snappy'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/snappypypy35/site-packages/snappy/__init__.py", line 1, in <module>
    from .snappy import (
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy.py", line 50, in <module>
    from .snappy_cffi import UncompressError, compress, decompress, \
  File "/home/ubuntu/snappypypy35/site-packages/snappy/snappy_cffi.py", line 1, in <module>
    from _snappy_cffi import ffi, lib
ImportError: No module named '_snappy_cffi'
```

**Comment by sormond** [comment]
Hi,
I am still getting this error despite the fix:

`>>> import snappy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sormond/anaconda3/envs/py27/lib/python2.7/site-packages/snappy/__init__.py", line 3, in <module>
    from .snappy import (
  File "/home/sormond/anaconda3/envs/py27/lib/python2.7/site-packages/snappy/snappy.py", line 51, in <module>
    from .snappy_cffi import UncompressError, compress, decompress, \
  File "/home/sormond/anaconda3/envs/py27/lib/python2.7/site-packages/snappy/snappy_cffi.py", line 3, in <module>
    from ._snappy_cffi import ffi, lib
ImportError: No module named _snappy_cffi
`

**Comment by darouwan** [comment]
fixed? It still not work in 0.5.3 for anaconda 3
